chall02/mhasan.c

```diff
@@ -0,0 +1,22 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+char *hv_rgb2hex(int r, int g, int b)
+{
+    char s1[3];
+    char *str;
+
+    if (r < 0 || g < 0 || b < 0 || r > 255 || g > 255 || b > 255)
+        return (NULL);
+    if (!(str = (char *)malloc(sizeof(char) * 8)))
+        return (NULL);
+    strcpy(str, "#");
+    sprintf(s1, "%02x", r);
+    strcat(str, s1);
+    sprintf(s1, "%02x", g);
+    strcat(str, s1);
+    sprintf(s1, "%02x", b);
+    strcat(str, s1);
```

Comment:
all of this could be written
```c
sprintf(s1, "#%02x%02x%02x", r, g, b);
```

Reply:
Could you please tell me what extra file I have pushed? I can't see anything extra in my local directory. I need to make sure it so that I won't do the same mistake in the future.

Reply:
Well, I think I found it.  It's Read.me file?

Reply:
yes